pydivide/read.py

```diff
@@ -12,49 +12,87 @@
 from .utilities import orbit_time
 from _collections import OrderedDict
 import builtins
+import re
 
 
-def read(input_time, instruments=None, insitu_only=False):
+def read(filename=None, input_time=None, instruments=None, insitu_only=False, specified_files_only=False):
     '''
     Read in a given filename in situ file into a dictionary object
     Optional keywords maybe used to downselect instruments returned
      and the time windows.
 
     Input:
-        filename: Name of the in situ KP file to read in.
-        time (Not Yet Implemeted): 
+        filename:
+            Name of the in situ KP file(s) to read in.
+        input_time:
             Set a time bounds/filter on the data
             (this will be necessary when this is called by a wrapper that
              seeks to ingest all data within a range of dates that may
              be allowed to span multiple days (files) ).
-        Instruments: (Not Yet Implemented)
+        instruments:
             Optional keyword listing the instruments to include 
             in the returned dictionary/structure.
+        insitu_only:
+            Optional keyword that allows you to specify that you only want 
+            to download insitu files.
+        specified_files_only:
+            Optional keyword that allows you to specify you only want filenames
+            given in 'filename' to be read in, not other files close in date/time
+            as well.
```

Comment:
This seemed the easiest way to me to allow the user an "out" if they truly only want specified filenames in "filename" to be read in, otherwise the default is to take the specified filename(s), read them in, but also read in any other filenames in the same time range as whatever specified files there are.